frontend/src/core/editor/components/EditorMainAction.test.js

```diff
@@ -0,0 +1,124 @@
+
+        expect(wrapper.find('.action-approve')).toHaveLength(1);
+        expect(wrapper.find('.action-suggest')).toHaveLength(0);
+        expect(wrapper.find('.action-save')).toHaveLength(0);
+
+        wrapper.find('.action-approve').simulate('click');
+
+        expect(updateStatus.calledWith(1, 'approve')).toBeTruthy();
+    });
+
+    it('renders the Approve button when translation is same as one in history', () => {
+        const params = {
+            initialTranslation: 'something',
+            isTranslator: true,
+            history: HISTORY,
+            translation: 'I was there before',
+        }
+        const wrapper = shallow(<EditorMainAction { ...params } />);
+
+        expect(wrapper.find('.action-approve')).toHaveLength(1);
+        expect(wrapper.find('.action-suggest')).toHaveLength(0);
+        expect(wrapper.find('.action-save')).toHaveLength(0);
+    });
+
+    it('renders the Suggest button when force suggestion is on', () => {
+        const params = {
+            isTranslator: false,
+            forceSuggestions: true,
+        }
+        const wrapper = shallow(<EditorMainAction { ...params } />);
+
+        expect(wrapper.find('.action-suggest')).toHaveLength(1);
+        expect(wrapper.find('.action-approve')).toHaveLength(0);
+        expect(wrapper.find('.action-save')).toHaveLength(0);
+    });
+
+    it('renders the Save button when force suggestion is off', () => {
+        const params = {
+            isTranslator: false,
+            forceSuggestions: false,
+        }
+        const wrapper = shallow(<EditorMainAction { ...params } />);
+
+        expect(wrapper.find('.action-save')).toHaveLength(1);
+        expect(wrapper.find('.action-suggest')).toHaveLength(0);
+        expect(wrapper.find('.action-approve')).toHaveLength(0);
+    });
+
+    it('renders the Save button when translation is not the same', () => {
+        const params = {
+            initialTranslation: 'something',
+            isTranslator: true,
+            history: HISTORY,
+            translation: 'something else',
+        }
+        const wrapper = shallow(<EditorMainAction { ...params } />);
+
+        expect(wrapper.find('.action-save')).toHaveLength(1);
+        expect(wrapper.find('.action-suggest')).toHaveLength(0);
+        expect(wrapper.find('.action-approve')).toHaveLength(0);
+    });
+
+    it('renders the Save button when same translation exists but is approved', () => {
+        const params = {
+            initialTranslation: 'something',
+            isTranslator: true,
+            history: HISTORY,
+            translation: 'hello, world!',
+        }
+        const wrapper = shallow(<EditorMainAction { ...params } />);
+
+        expect(wrapper.find('.action-save')).toHaveLength(1);
+        expect(wrapper.find('.action-suggest')).toHaveLength(0);
+        expect(wrapper.find('.action-approve')).toHaveLength(0);
+    });
+
+    it('renders the Save button when user does not have permission', () => {
```

Comment:
I think we have a bug (here and in prod): we should show the "Suggest" button for non-privileged users.

https://mozilla-l10n.github.io/localizer-documentation/tools/pontoon/translate.html#submit-a-suggestion

Reply:
Nice catch! Thank you unit tests! :D